src/solvers/dg/dg_2d.jl

```diff
@@ -270,9 +276,45 @@ function calc_volume_integral!(du::AbstractArray{<:Any,4}, u,
                                nonconservative_terms, equations,
                                volume_integral::VolumeIntegralFluxDifferencing,
                                dg::DGSEM, cache)
+
+  @unpack weights = dg.basis
+  @unpack du_ec, du_cen = cache
+
+  # du_ec = similar(du)
+  du_ec .= 0.0
+  # du_cen = similar(du)
+  du_cen .= 0.0
+
   @threaded for element in eachelement(dg, cache)
-    split_form_kernel!(du, u, nonconservative_terms, equations, volume_integral.volume_flux, dg, cache, element)
+    # compute volume integral with flux, and for comparison with central flux
+    split_form_kernel!(du_ec, u, nonconservative_terms, equations, volume_integral.volume_flux, dg, cache, element)
+    split_form_kernel!(du_cen, u, nonconservative_terms, equations, flux_central, dg, cache, element)
+
+    # compute entropy production of both volume integrals
+    delta_entropy = 0.0
```

Comment:
```suggestion
    delta_entropy = zero(eltype(u))
```
That should be better w.r.t. type stability (e.g. for AD)